### PR TITLE
PLAT-78973: Updated Icon to support arbitrary strings as icon names

### PR DESCRIPTION
**Pull request by Djspaceg**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed

* [ ] This is an API breaking change

### Issue Resolved / Feature Added
Icon names longer than one character that are outside of the iconsList were being treated as URLs.

### Resolution
Updated the checker code to look for URL-like symbols to determine "URL-ness" of a string.

### CODE REVIEW

File: packages/ui/Icon/Icon.js [hunk 1]
```diff
@@ -188,8 +186,8 @@ const Icon = kind({
 					} else if (iconProp.indexOf('0x') === 0) {
 						// Converts a hex reference in string form
 						icon = String.fromCodePoint(iconProp);
-					} else if (isSingleCharacter(iconProp)) {
-						// A single character is assumed to be an explicit icon string
+					} else if (!isUri(iconProp)) {
```
Comment (ryanjduffy) [inline packages/ui/Icon/Icon.js]: Let's add some unit tests the valid `icon` values since we're updating this.

Comment (Djspaceg) [reply packages/ui/Icon/Icon.js]: Added several tests, which include testing this new approach.

**Comment by Djspaceg** [comment]
Added a bunch of tests to validate the new code is working and catches edge-cases.
